dxp/sennajs_integration.md

```diff
@@ -59,6 +59,22 @@ These paths, are visible in the global `Liferay.SPA.excludedPaths` variable.
 
 By default Senna.js in DXP will contain the required parameters needed by portlets, specifically `p_p_id`, `p_p_lifecycle`, `p_p_state` and `p_p_auth`.
 
+## Events
+
+By default, Senna.js triggers some events during the navigation process that can be used to remove elements or to destroy events from the page application, in DXP the events are transmitted via `Liferay.on`.
+
+-   `beforeNavigate`
+-   `beforeScreenFlip`
+-   `endNavigate`
+-   `screenActivate`
+-   `screenDeactivate`
+-   `screenDispose`
+-   `screenFlip`
+-   `screenLoad`
+-   `startNavigate`
```

Comment:
I'd be interested in knowing some more about what each of these does.

For example, without looking at the code, I don't know the difference between `beforeNavigate`, `startNavigate` and `beforeScreenFlip` (although I can make a guess, and make assumptions about the order they fire in).

Likewise, I don't know the order of or difference between `screenDeactivate` and `screenDispose` — again, I can take a guess, or look at the code, but it would be cool if it were spelled out for me.

Or another way to think of it, we might want to define what the basic terms mean:

- What "flipping" a screen means.
- What "activating" and "deactivating" (and "disposing") a screen means.

Of the terms, the only ones that have relatively clear meanings are "navigate" and "load", but even then we need to be clear about what they mean in the context of Senna.

Reply:
Agreed... a high level description of the lifecycle events would be helpful. We have some information in the source code itself or in sennajs.com, so just putting together some of it here should be enough, I think.